scopes.md

```diff
@@ -0,0 +1,146 @@
+
+Granted scopes may differ from what was requested by the client.
+
+[OpenID Connect Core 1.0][connect] defines a few scopes which are specific for getting the authorized user information. API clients must include the `openid` scope to the scope request list. The `openid` scope gives access to using OpenID Connect authentication and if missing the API authentication will fail.
+
+diaspora* also supports the `profile` and `email` scopes according to [OpenID Connect Core 1.0][connect-scopes].
+
+Other scopes are diaspora* API specific. All scopes besides `openid` are optional.
+
+There is a special case scope `public:read` which is always granted even if not explicitly required. The `public:read` scope gives access to data which is public anyway and sometimes available even with no authorization at all, so it is available to every authorized client too, there is no reason to allow opt out for it.
+
+Also it is possible that scopes depend on other scopes. It means that requesting and granting of a specific scope is only possible along with the scope it depends on. For example the `private:read` and `private:modify` scope which grants reading private posts permission depends on `contacts:read` scope.
+
+Each API endpoint requires at least one scope to be granted. If the access token used for making a request wasn't granted the scope of the respective endpoint then the response will have a `403` status code and no actual content.
+
+In this specification each endpoint description contains the name of the scope required to access this endpoint. Most of the time this is only one scope. However for some types of data like posts the required scope depends on the data visibility (public vs private). Besides this, for some endpoints, like streams the dataset in response depends on the set of granted scopes.
+
+## Scopes
+
+### contacts:read
+
+This scope gives access to the following requests:
+- Read aspect list
+- Read aspects membership
+- Read private user profiles of contacts
+- Read contacts of contacts when allowed by contact's aspect setting
+- User search will include contacts who are hidden from public search
+
+### contacts:modify
+
+This scope gives access to the following requests:
+- Add new aspects
+- Rename aspects and change aspects properties
+- Delete aspects
+- Add/remove person to/from aspects
+
+### conversations
+
+This scope gives access to private messaging, i.e. the following requests:
+- Create conversations
+- Send private messages
+
+### email
+
+This scope is defined by the OpenID Connect spec. It allows reading user's own email address using the UserInfo route or from the ID Token.
+
+### interactions
+
+This scope gives access to the following requests:
+- Create/delete comments
+- Report posts/comments
+- Post/remove likes
+- Subscribe/mute posts
+- Hide posts from streams
```

Comment:
there is a voting method now for polls, which I think of as an interaction with a post so would we want to add that to the list here?